modules/andes-core/broker/src/main/java/org/wso2/andes/thrift/MBThriftServer.java

```diff
@@ -30,13 +30,19 @@
 import org.wso2.andes.thrift.slot.gen.SlotManagementService;
 
 import java.net.InetSocketAddress;
+import java.util.concurrent.TimeUnit;
 
 /**
- * This class is take cares of starting and stopping the thrift server which is used to do slot
- * communication.
+ * This class takes care of starting and stopping the thrift server which is used to do slot communication.
  */
 public class MBThriftServer {
 
+    /**
+     * Default thrift server request timeout. Server stop() will also wait this amount of time at max before
+     * returning. The time unit is seconds.
+     */
+    private static final int THRIFT_SERVER_REQUEST_TIMEOUT = 20;
```

Comment:
We can make it a broker config if we want. But I don't think it will be very  useful for a user.